src/rlimitfloatenv.cpp

```diff
@@ -0,0 +1,82 @@
+#include <fenv.h>
+#include <sys/time.h>
+#include <sys/resource.h>
+#include "shareddata.h"
+#include "config.h"
+#include "dmtcp.h"
+
+/*************************************************************************
+ *
+ *  Save and restore rlimit and float exception settings.
+ *
+ *************************************************************************/
+
+namespace dmtcp
+{
+
+static int roundingMode = -1;
+static fenv_t envp;
+static rlim_t rlim_cur_nofile = 0;
+static rlim_t rlim_cur_stack = 0;
+
+static void
+save_rlimit_float_settings()
+{ roundingMode = fegetround();
+  fegetenv(&envp);
+  { struct rlimit rlim = {0, 0};
+    getrlimit(RLIMIT_NOFILE, &rlim);
+    rlim_cur_nofile = rlim.rlim_cur;
+    getrlimit(RLIMIT_STACK, &rlim);
+    rlim_cur_stack = rlim.rlim_cur;
+  }
+}
+
+static void
+restore_rlimit_float_settings()
+{ fesetenv(&envp);
+  fesetround(roundingMode);
+  { struct rlimit rlim = {0, 0};
+    getrlimit(RLIMIT_NOFILE, &rlim);
+    rlim.rlim_cur = rlim_cur_nofile;
+    setrlimit(RLIMIT_NOFILE, &rlim);
```

Comment:
Should we be checking the return code for failure? Another check would be to see if the current hard limit is lower than rlim_cur_nofile. Similarly for stack.

Reply:
I'll check and fix the style issues.  And I'll also see if Travis continues to give a problem with the 'java1' test.

By the way, I just now pushed in one more change.  I'm renaming `ProcessInfo::refresh()` to `ProcessInfo::getState()`.  That seems to be more meaningful, given the several calls to `getXXX()` in the implementation of the method.